src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -0,0 +1,558 @@
+ */
+public abstract class SFBaseSession {
+  private final Properties clientInfo = new Properties();
+  private final AtomicBoolean autoCommit = new AtomicBoolean(true);
+  // Injected delay for the purpose of connection timeout testing
+  // Any statement execution will sleep for the specified number of milliseconds
+  private final AtomicInteger _injectedDelay = new AtomicInteger(0);
+  // Connection properties map
+  private final Map<SFSessionProperty, Object> connectionPropertiesMap = new HashMap<>();
+  // Custom key-value map for other options values *not* defined in SFSessionProperties,
+  // i.e., session-implementation specific
+  private final Map<String, Object> customSessionProperties = new HashMap<>(1);
+  protected List<SFException> sqlWarnings = new ArrayList<>();
+  // Unique Session ID
+  private String sessionId;
+  // Database versions
+  private String databaseVersion = null;
+  private int databaseMajorVersion = 0;
+  private int databaseMinorVersion = 0;
+  // Used for parsing results
+  private SnowflakeType timestampMappedType = SnowflakeType.TIMESTAMP_LTZ;
+  private boolean isResultColumnCaseInsensitive;
+  private boolean isJdbcTreatDecimalAsInt = true;
+  private boolean treatNTZAsUTC;
+  private boolean preparedStatementLogging = false;
+  // Inject failure for testing
+  private String injectFileUploadFailure;
+  private boolean enableHeartbeat;
+  private boolean formatDateWithTimezone;
+  private boolean enableCombineDescribe;
+  private boolean clientTelemetryEnabled = false;
+  private boolean useSessionTimezone;
+  // The server can read array binds from a stage instead of query payload.
+  // When there as many bind values as this threshold, we should upload them to a stage.
+  private int arrayBindStageThreshold = 0;
+  private boolean storeTemporaryCredential;
+  private String serviceName;
+  private boolean sfSQLMode;
+  // whether to enable conservative memory usage mode
+  private boolean enableConservativeMemoryUsage;
+  // the step in MB to adjust memory usage
+  private int conservativeMemoryAdjustStep = 64;
+  private int clientMemoryLimit;
+  private int clientResultChunkSize;
+  private int clientPrefetchThreads;
+  // validate the default parameters by GS?
+  private boolean validateDefaultParameters;
+  // Current session context w/ re to database, schema, role, warehouse
+  private String database;
+  private String schema;
+  private String role;
+  private String warehouse;
+  // For Metadata request(i.e. DatabaseMetadata.getTables or
+  // DatabaseMetadata.getSchemas,), whether to use connection ctx to
+  // improve the request time
+  private boolean metadataRequestUseConnectionCtx = false;
+  // For Metadata request(i.e. DatabaseMetadata.getTables or
+  // DatabaseMetadata.getSchemas), whether to search using multiple schemas with
+  // session database
+  private boolean metadataRequestUseSessionDatabase = false;
+
+  public Properties getClientInfo() {
```

Comment:
please add javadoc for at least all public methods. https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required

Reply:
Will add this.

Reply:
I added a few comments where attributes are shared among different implementations of `SFBaseSession`. However, most of these methods are simple getters and setters on session properties. Most of them were ported over from their original place in `SFSession`, where they did not have original comments/Javadocs, so I actually don't know how most of these are used.